src/Catalog.GraphQL/Catalog/CatalogQueryType.cs

```diff
@@ -0,0 +1,22 @@
+﻿using Catalog.Abstractions;
+using Catalog.GraphQL.Models;
+using Microsoft.EntityFrameworkCore;
+
+namespace Catalog.GraphQL.Catalog;
+
+[ExtendObjectType("Query")]
+public class CatalogQueryType
+{
+    [UsePaging]
+    [UseProjection]
+    [UseFiltering]
+    public Task<List<CatalogDto>> GetCatalogs([Service] IApplicationDbContext context) =>
+        context.Catalogs.Select(c => CatalogDto.FromCatalog(c)).ToListAsync();
```

Comment:
Is it possible to use IQueryable instead of the List? I'm wondering how would this paging and working work.

Reply:
Pagination works with IEnumerable and IQueryable. In the case of IQueryable- the pagination operations can be directly translated to native database queries

Reply:
Is this an Application level? Currently looks like UI layer.

Anyway, to be frank i would allow to expose IQueryable in this case. It's much better when the query is generated from the database, instead taking everything from the database and then doing all this pagination and filtering.

Reply:
I exposed the IQuerable with entity types. The client could fetch information all it wants.  Here are some examples of generated queries:
1. Get the list of catalogs with children's.
![image](https://user-images.githubusercontent.com/2194278/209444074-63383ffc-0a80-4810-b600-7e7a21bf9c26.png)
Result query:
![image](https://user-images.githubusercontent.com/2194278/209444084-3120a0e0-484d-4ceb-ba86-c93e418efcda.png)

2. Get the list of catalogs with children's and products:
![image](https://user-images.githubusercontent.com/2194278/209444118-01ea5269-0b9a-4198-8eb4-88a092a2c6d4.png)
Result query
![image](https://user-images.githubusercontent.com/2194278/209444128-df43c555-d2c0-4d7f-83fd-db83808f3ef8.png)

Reply:
Here is another case pf generated query.

Reply:
![image](https://user-images.githubusercontent.com/2194278/209726297-92362b0b-b1b4-4f76-ae04-a6089b76e269.png)
![image](https://user-images.githubusercontent.com/2194278/209726304-c40dc2a8-9ace-4801-b573-3d9f112de415.png)